File: rb/features.rb
# YandereApiV2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module YandereApiV2Features
  def self.make_feature(name)
    case name
    when "base"
      YandereApiV2BaseFeature.new
    when "ratelimit"
      YandereApiV2RatelimitFeature.new
    when "retry"
      YandereApiV2RetryFeature.new
    when "test"
      YandereApiV2TestFeature.new
    when "timeout"
      YandereApiV2TimeoutFeature.new
    else
      YandereApiV2BaseFeature.new
    end
  end
end

File: py/features.py
# YandereApiV2 SDK feature factory

from feature.base_feature import YandereApiV2BaseFeature
from feature.test_feature import YandereApiV2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: YandereApiV2BaseFeature(),
        "test": lambda: YandereApiV2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
